mycompany-seeq-link-connector-myconnector/src/main/java/com/mycompany/seeq/link/connector/DatasourceSimulator.java

```diff
@@ -30,55 +94,177 @@ public String getName() {
         public boolean getStepped() {
             return this.stepped;
         }
+
+        public static class Value {
+            private TimeInstant timestamp;
+            private double measure;
+
+            public Value(TimeInstant timestamp, double value) {
+                this.timestamp = timestamp;
+                this.measure = value;
+            }
+
+            public TimeInstant getTimestamp() {
+                return this.timestamp;
+            }
+
+            public double getMeasure() {
+                return this.measure;
+            }
+        }
+    }
+
+    public class Constant {
+        private String id;
+        private String name;
+        private String unitOfMeasure;
+        private Object value;
+
+        public Constant(String elementId, int constantId, String unitOfMeasure, Object value) {
+            this.id = String.format("Element=%s;Constant=%d", elementId, constantId);
+            this.name = String.format("Simulated Constant #%d", constantId);
+            this.unitOfMeasure = unitOfMeasure;
+            this.value = value;
+        }
+
+        public String getId() {
+            return this.id;
+        }
+
+        public String getName() {
+            return this.name;
+        }
+
+        public String getUnitOfMeasure() {
+            return unitOfMeasure;
+        }
+
+        public Object getValue() {
+            return value;
+        }
     }
 
-    public DatasourceSimulator(int tagCount, Duration signalPeriod) {
-        this.tagCount = tagCount;
+    // To be able to yield consistent, reproducible tag values, we need a constant seed. This helps us
+    // approximate the behaviour of a real datasource which should be deterministic.
+    private final int RandomnessSeed = 1_000_000;
+
+    private final Random RNG = new Random(RandomnessSeed);
+
+    private boolean connected;
+    private Duration signalPeriod;
```

Comment:
To follow Java's conventions I think it would be better to:
- Move all fields to the top of the class
- Make `RandomnessSeed` `static` and rename it to `RANDOMNESS_SEED`
- Make `RNG` static
- Move all inner classes to the bottom of the class